### ISSUE TITLE
FVA issue

### ISSUE THREAD

**Issue by cvitlanko** [opencobra/cobrapy#700]
I have 0.11.3 cobra version. I performed FVA 

```fva_res = flux_variability_analysis(chos, fraction_of_optimum=0.9, loopless=True)```

I wanted to see in which reactions does metabolite participate
```chos.metabolites.get_by_id("glc_D_c").summary(solution=fva_res)```

and got this error. 
```---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-70-fc2e0d7a7a65> in <module>()
----> 1 chos.metabolites.get_by_id("glc_D_c").summary(solution=fva_res_s)

~/anaconda3/lib/python3.6/site-packages/cobra/core/metabolite.py in summary(self, solution, threshold, fva, floatfmt)
    239         from cobra.flux_analysis.summary import metabolite_summary
    240         return metabolite_summary(self, solution=solution, threshold=threshold,
--> 241                                   fva=fva, floatfmt=floatfmt)
    242 
    243     def _repr_html_(self):

~/anaconda3/lib/python3.6/site-packages/cobra/flux_analysis/summary.py in metabolite_summary(met, solution, threshold, fva, floatfmt)
     48     for rxn in met.reactions:
     49         rxn_id.append(format_long_string(rxn.id, 10))
---> 50         flux.append(solution.fluxes[rxn.id] * rxn.metabolites[met])
     51         reaction.append(format_long_string(rxn.reaction, 40 if fva else 50))
     52 

~/anaconda3/lib/python3.6/site-packages/pandas/core/generic.py in __getattr__(self, name)
   3612             if name in self._info_axis:
   3613                 return self[name]
-> 3614             return object.__getattribute__(self, name)
   3615 
   3616     def __setattr__(self, name, value):

AttributeError: 'DataFrame' object has no attribute 'fluxes'
```

if I run

```chos.metabolites.get_by_id("glc_D_c").summary(fva= 0.9)```

I get the following error

```---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-74-fd6cf7906b0f> in <module>()
----> 1 chos.metabolites.get_by_id("glc_D_c").summary(fva= 0.9)

~/anaconda3/lib/python3.6/site-packages/cobra/core/metabolite.py in summary(self, solution, threshold, fva, floatfmt)
    239         from cobra.flux_analysis.summary import metabolite_summary
    240         return metabolite_summary(self, solution=solution, threshold=threshold,
--> 241                                   fva=fva, floatfmt=floatfmt)
    242 
    243     def _repr_html_(self):

~/anaconda3/lib/python3.6/site-packages/cobra/flux_analysis/summary.py in metabolite_summary(met, solution, threshold, fva, floatfmt)
     56     if fva:
     57         fva_results = flux_variability_analysis(
---> 58             met.model, met.reactions, fraction_of_optimum=fva)
     59 
     60         flux_summary.index = flux_summary["id"]

~/anaconda3/lib/python3.6/site-packages/cobra/flux_analysis/variability.py in flux_variability_analysis(model, reaction_list, loopless, fraction_of_optimum, pfba_factor)
     85         reaction_list = model.reactions
     86     else:
---> 87         reaction_list = model.reactions.get_by_any(reaction_list)
     88 
     89     prob = model.problem

~/anaconda3/lib/python3.6/site-packages/cobra/core/dictlist.py in get_by_any(self, iterable)
     90         if not isinstance(iterable, list):
     91             iterable = [iterable]
---> 92         return [get_item(item) for item in iterable]
     93 
     94     def query(self, search_function, attribute=None):

~/anaconda3/lib/python3.6/site-packages/cobra/core/dictlist.py in <listcomp>(.0)
     90         if not isinstance(iterable, list):
     91             iterable = [iterable]
---> 92         return [get_item(item) for item in iterable]
     93 
     94     def query(self, search_function, attribute=None):

~/anaconda3/lib/python3.6/site-packages/cobra/core/dictlist.py in get_item(item)
     86                 return item
     87             else:
---> 88                 raise TypeError("item in iterable cannot be '%s'" % type(item))
     89 
     90         if not isinstance(iterable, list):

TypeError: item in iterable cannot be '<class 'frozenset'>'```

**Comment by Midnighter** [comment]
Hi Svetlana,

The first error occurs because summary expects a normal `cobra.Solution` instance and not an FVA result.

```python
import cobra.flux_analysis as cflux

sol = cflux.pfba(model)
model.metabolites.glu__L_c.summary(solution=sol)
```

```
PRODUCING REACTIONS -- L-Glutamate (glu__L_c)
---------------------------------------------
%       FLUX  RXN ID      REACTION
----  ------  ----------  --------------------------------------------------
100%   4.54   GLUDy       glu__L_c + h2o_c + nadp_c <=> akg_c + h_c + nad...

CONSUMING REACTIONS -- L-Glutamate (glu__L_c)
---------------------------------------------
%       FLUX  RXN ID      REACTION
----  ------  ----------  --------------------------------------------------
95%    4.32   Biomass...  1.496 3pg_c + 3.7478 accoa_c + 59.81 atp_c + 0....
5%     0.223  GLNS        atp_c + glu__L_c + nh4_c --> adp_c + gln__L_c +...
```

The second error that you report is definitely a bug. I'll dig into it and also I think it would be nice to pass in an existing FVA solution into a summary.

**Comment by stairs** [comment]
@Midnighter Looks like this bug was fixed with the update on June 6th ( #718). `flux_variability_analysis` function is designed to work with a list but frozenset was passed in FVA scenario. The issue can be closed now.

**Comment by Midnighter** [comment]
Great :smiley:
